app/controllers/topics_controller.rb

```diff
@@ -8,6 +8,38 @@ def index
     end
   end
 
+  def vote
+    @topic = Topic.find params[:id]
+    @topic.increment :votes
+    respond_to do |format|
+      if @topic.save 
```

Comment:
By atomic I mean that if you have shit for luck, two people will hit the same controller at the same time and one of the votes will be lost. Here's what happens:

1. Johannes hits the vote button first because he's way cool and has lightning reflexes.
2. The controller loads the topic from the database and into memory for Johannes' request. It has 0 votes in the database at this point, and so the model instance that is loaded into memory also has 0 votes.
3. Ole Henrik thinks that topic is really awesome, too, and so he votes on it as well.
4. The controller load the topic from the database and into memory for Ole Henrik's request. It still has 0 votes in the database because the controller didn't finish processing Johannes' request yet, and so the model instance that is loaded into memory also has 0 votes.
5: The controller finishes processing Johannes' request, which adds 1 to the number of votes that the model instance has _in memory_ (which is 0). It saves it to the database. The topic now has 1 vote in the database.
6: The controller continues processing Ole Henrik's request, which adds 1 to the number of votes that the model instance has _in memory_ (which is still 0, because it was loaded from the database before Johannes' request finished processing). It saves it to the database. The topic now has 1 vote in the database.

Oops, we lost a vote!